File: CODEFORCES/B_Ropes.cpp
#include <bits/stdc++.h>
using namespace std;
#define pb push_back
double n, k;
vector<double> v;

bool possible(double m)
{

    if (m == 0)
        return true;
    int ans = 0;
    for (int i = 0; i < n; i++)
    {
        ans += (v[i] / m);
    }
    return ans >= k;
}

signed main()
{
    cin >> n >> k;
    for (int i = 0; i < n; i++)
    {
        double temp;
        cin >> temp;
        v.pb(temp);
    }
    double l = 0, r = 1e8, eps = 1e-7;
    cout << setprecision(7);
    double res = -1;
    while ((r - l) > eps)
    {
        double mid = l + (r - l) / 2;
        if (possible(mid))
        {
            l = mid;
            res = mid;
        }
        else
            r = mid;
    }
    cout << res;
}

File: CODEFORCES/B_Ropes.py
def func(length):
    # print(lst)
    sum = 0
    for i in lst:
        sum += (i // length)
        
    return sum >= k

n, k = map(int, input().split())
lst = []
for i in range(n):
    temp = int(input())
    lst.append(temp)
    
# print(lst)
l = 0
r = 1e8

res = -1

while ((r - l) > (1e-7)):
    mid = l + (r - l) / 2
    if func(mid):
        res = max(res, mid)
        l = mid 
    else:
        r = mid
        
print(res)
    